File: boredom/The Inventory Lab/inventorylab.py
from flask import Flask, request
from flask_cors import CORS
from inventorylabdb import addstuff as pooff, getstuff, updating, delete

app = Flask(__name__)  # Creates an instance of app
CORS(app, origins=[
    "http://localhost:63342"

])

@app.route('/addstuff', methods=['POST'])
def addstuff():
    data = request.get_json()

    item = data.get('item')
    price = data.get('price')
    quantity = data.get('quantity')
    print(type(item), type(price), type(quantity))

    if item == '' or item.isspace() == True:
        return 0
    elif price == 0 or price.isspace() == True:
        return 0
    elif quantity == 0 or quantity.isspace() == True:
        return 0
    a = pooff(item, price, quantity)
    return str(a)

@app.route('/getstuff', methods=['GET'])
def readthetable():
    lst = []
    aaa = getstuff()
    print(aaa)
    print(aaa[0])
    print(aaa[0][1])
    for i in range (len(aaa)):
        dict = {
            "id":aaa[i][0],
            "item":aaa[i][1],
            "price":aaa[i][2],
            "quantity": aaa[i][3]
        }
        lst.append(dict)
    print(lst)
    return lst

@app.route('/stuff/<int:id>', methods=['PUT'])
def update(id):
    data = request.get_json()

    NewQuantity = data.get('NewQuantity')

    id = updating(id, NewQuantity)

    return str(id)

@app.route('/stuff/<int:id>', methods=['DELETE'])
def deleting(id):
    id = delete(id)

    return str(id)

if __name__ == '__main__':
    app.run(host='0.0.0.0', debug=True, port=5006)

File: boredom/The Inventory Lab/inventorylab.js


function addstuff(){
    let item = document.getElementById('item').value;
    let price = document.getElementById('price').value;
    let quantity = document.getElementById('quantity').value;
    const url = `http://127.0.0.1:5006/addstuff`;

    fetch(url, {
        method: "POST",
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ item:item, price:price, quantity:quantity })
    })

    .then(response => response.ok ? response.json() : response.text().then(Promise.reject))
    .then(data => {
    console.log(data)
    location.reload()

 });
}


function gettingstuff(){
 const url = `http://127.0.0.1:5006/getstuff`;
    fetch(url, {
        method: "GET",
        headers: { 'Content-Type': 'application/json' }
    })
    .then(response => response.ok ? response.json() : response.text().then(Promise.reject))
    .then(data => {
    console.log(data)

    let displaylist = '  <tr> <th>Item</th> <th>Price</th> <th>Quantity</th> <th>New Quantity</th> <th>Save</th> <th>Delete</th> </tr>'
    for (let i in data){
        let checked = ''
        const id = data[i]["id"]
        const item = data[i]["item"]
        const price = data[i]["price"]
        const quantity = data[i]["quantity"]
        // displaylist += `<li><input onChange=checkboxclicked(${id}) type="checkbox" ${checked}/>${data[item]["Task"]}&nbsp;&nbsp;<button onclick="deleting(${id})">Delete</button></li>`
        displaylist += `<tr> <td>${item}</td> <td>${price}</td> <td>${quantity}</td> <td><input id='${id}' type='number'/></td> <td><button onclick="update(${id})">Update</button></td> <td><button onclick="deleting(${id})">Delete</button></td> </tr>`
        }
    document.getElementById('rowsappendkarnihai').innerHTML = displaylist;
});

    }

window.addEventListener("load", gettingstuff());

function update(id){
    const NewQuantity = document.getElementById(id).value;
    console.log(id)
   const url = `http://127.0.0.1:5006/stuff/${id}`
 fetch(url, {
        method: "PUT",
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({'NewQuantity':NewQuantity})
    })
    .then(response => response.ok ? response.json() : response.text().then(Promise.reject))
    .then(data => {
    console.log(data)
    gettingstuff()
 });


}

function deleting(id){
    console.log(id)
   const url = `http://127.0.0.1:5006/stuff/${id}`
 fetch(url, {
        method: "DELETE",
        headers: { 'Content-Type': 'application/json' }
    })

.then(response => response.ok ? response.json() : response.text().then(Promise.reject))
    .then(data => {
    console.log(data)
    gettingstuff()
});
}
